src/net.cpp

```diff
@@ -1136,6 +1152,31 @@ void CConnman::CreateNodeFromAcceptedSocket(SOCKET hSocket,
         }
     }
 
+	// Currently, users can open multiple connections to a reachable node using the same IP address and a 
+	// different port and request blockchain data at each connection. This behavior has the following impacts:
+	//   (1) Overwhelm the reachable node and consume its bandwidth by requesting data and discarding it
+	//     (see video demonstrations here: https://www.dropbox.com/s/pvasua7lnepdgfq/Occupy_Connections.mp4?dl=0
+	//     and https://www.dropbox.com/s/45ld1h7cxi3xzj6/RequestingBlockchain.mp4?dl=0)
+	//   (2) Undermine the transaction relaying anonymity since multiple connections to each node increase the
+	//     probability of mapping a transaction to an IP address (currently, the random delay factor in the
+	//     trickling effect implicitly assumes the attacker has one connection to each node). By establishing
+	//     multiple connections, the attacker undermines the anonymity property. 
```

Comment:
This is untrue. All inbound connections have the same relay time assigned. See https://github.com/bitcoin/bitcoin/blob/c6b30ccb2eee5f80f844f79766591f0a1326ce43/src/net.cpp#L3014

Having multiple connections should be no different than a single one.

Reply:
Feasibly occupying multiple incoming connections using the same IP address will undermine the transaction anonymity property provisioned by the "diffusion" process. In the current deployment, a node announces a transaction to each connection after a certain delay. The assumption is that the adversary will not be able to infer the source of the transaction as it might receive the INV from another connection rather than the source (ref diffusion and exponential delay). 

If an adversary occupies multiple incoming connections, the adversary can easily infer the source of the transaction, despite that delay (higher probability of receiving INV from the source rather than any other connection of that source). Allowing multiple incoming connections from the same IP address makes it a lot easier.

Reply:
That's not correct. All incoming connections share the same Poisson timer for diffusion, so an attacker that creates multiple incoming connections should not be able to observe (significantly) more than with a single connection.

Reply:
Thanks Pieter. A test case would be to (1) set up a reachable node (say A), (2) find its outgoing connections (getpeerinfo), (3) establish one connection to each of A's outgoing connections, (4) establish multiple connections to A (say 30), and (5) generate a transaction from A and see how much can be inferred. I can do this experiment on my nodes and get back to you with the results. 

You have pointed out that that attacker will not be able to observe (significantly) more than with a single connection. Does significantly mean a probability in k where k is the total number of incoming connections?

Reply:
@beingsmaad That would be interesting to get data on!

The only gain you should get from having multiple incoming connections vs. just a single one is avoiding more internal latency/processing delay in the message processing loop. I.e., if CPU speed was infinite, there should be no gain at all.

Reply:
@sipa I will get some data on it and share it with you (possibly over the weekend). Will generate a few transactions in the testnet and see how much can be inferred by making 2-30 incoming connections.